src/Workspaces/Core/Portable/Storage/SQLite/v2/SQLitePersistentStorageConstants.cs

```diff
@@ -67,9 +68,12 @@ internal static class SQLitePersistentStorageConstants
         /// The format of the table is:
         ///
         ///  ProjectData
-        ///  -------------------------------------------------------------------
-        ///  | DataId (primary key, integer) | | Checksum (blob) | Data (blob) |
-        ///  -------------------------------------------------------------------
+        ///  ------------------------------------------------------------------------------------------------
+        ///  | ProjectPathId (int) | ProjectNameId (int) | DataNameId (int) | Checksum (blob) | Data (blob) |
+        ///  ------------------------------------------------------------------------------------------------
+        ///  | Primary Key                                                  |
+        ///  ----------------------------------------------------------------
```

Comment:
this (and below) is the core db schema idea that is being applied here.  previously we used a separate table (the string table) to creating a mapping of these 3 ints to a final int used as the key in this table.  now we just use the 3 ints directly in this table instead.